core_api/src/routes/file.py

```diff
@@ -115,10 +117,29 @@ def get_file(file_uuid: UUID, user_uuid: Annotated[UUID, Depends(get_user_uuid)]
 
     Returns:
         File: The file
+
+    Raises:
+        404: If the file isn't found, or the creator and requester don't match
     """
-    file = storage_handler.read_item(file_uuid, model_type="File")
+    not_found = JSONResponse(
```

Comment:
Nitpick - you could move the creation of this object into a function - `def file_not_found_response(file_uuid: UUID) -> JSONResponse` or the like. Then, don't call this function yet...